platform-jobs/samza/course-batch-updater/src/main/java/org/sunbird/jobs/samza/util/BatchStatusUtil.java

```diff
@@ -64,18 +65,18 @@ public static void updateCompletedBatch() {
             }};
             List<Row> rows = SunbirdCassandraUtil.read(keyspace, table, dataToSelect);
             if(CollectionUtils.isNotEmpty(rows)) {
-                List<String> batchIds = new ArrayList<>();
+                Map<String, String> batchIdCourseIdMap = new HashMap<>();
                 for (Row row : rows) {
                     if (StringUtils.isNotBlank(row.getString("enddate"))) {
                         Date startDate = format.parse(row.getString("enddate"));
                         if (currentDate.compareTo(startDate) >= 0) {
-                            batchIds.add(row.getString("batchid"));
+                            batchIdCourseIdMap.put(row.getString("courseid"), row.getString("batchId"));
```

Comment:
Use `batchId` as Key. There is a chance of one course having multiple `batchIds`. In this case, it will overwrite the data.